code/obj/item/spirit_candle.dm

```diff
@@ -0,0 +1,132 @@
+//Your offensive weapon against wraiths and invisible folks
+/obj/item/device/light/spirit_candle
+	name = "spirit candle"
+	desc = "Feels vaguely ominous!"
+	icon = 'icons/obj/items/skull_candle.dmi'
+	icon_state = "unmelted-unlit"
+	w_class = W_CLASS_SMALL
+	density = 0
+	anchored = 0
+	opacity = 0
+	col_r = 0.2
+	col_g = 0.2
+	col_b = 0.8
+	var/sparks = 7
+	var/burn_state = 0
+	var/burnt = false
+	var/light_ticks = 60
+
+	New()
+		..()
+
+	attack_self(mob/user)
+		if (src.on)
+			src.visible_message("<span class='notice'>[user] blows on [src], its eyes emit a threatening glow!</span>")
+			for(var/mob/wraith/W in orange(4, user))
+				if (W.justmanifested <= 0)
+					SPAWN(5 DECI SECOND)	//Give them half a second to flee
+						W.makeCorporeal()
+						W.forced_manifest = TRUE
+						boutput(W, "<span class='alert'>A malignant spirit pulls you into the physical world! You begin to gather your forces to try and escape to the spirit realm...</span>")
+						SPAWN(30 SECOND)
+							W.makeIncorporeal()
+							W.justmanifested = 5	//To avoid spamming the candle
+							W.forced_manifest = FALSE
+			var/turf/T = get_turf(src)
+			playsound(src.loc, "sound/voice/chanting.ogg", 50, 0)
+			new /obj/overlay/darkness_field(T, 10 SECOND, radius = 5.5, max_alpha = 250)
+			new /obj/overlay/darkness_field{plane = PLANE_SELFILLUM}(T, 10 SECOND, radius = 5.5, max_alpha = 250)
+			src.put_out(user)
+
+	attackby(obj/item/W, mob/user)
+		if (!src.on && !burnt)
+			if (isweldingtool(W) && W:try_weld(user,0,-1,0,0))
+				src.light(user, "<span class='alert'><b>[user]</b> casually lights [src] with [W], what a badass.</span>")
+
+			else if (istype(W, /obj/item/clothing/head/cakehat) && W:on)
+				src.light(user, "<span class='alert'>Did [user] just light \his [src] with [W]? Holy Shit.</span>")
+
+			else if (istype(W, /obj/item/device/igniter))
+				src.light(user, "<span class='alert'><b>[user]</b> fumbles around with [W]; sparks erupt from [src].</span>")
+
+			else if (istype(W, /obj/item/device/light/zippo) && W:on)
+				src.light(user, "<span class='alert'>With a single flick of their wrist, [user] smoothly lights [src] with [W]. Damn they're cool.</span>")
+
+			else if ((istype(W, /obj/item/match) || istype(W, /obj/item/device/light/candle)) && W:on)
+				src.light(user, "<span class='alert'><b>[user] lights [src] with [W].</span>")
+
+			else if (W.burning)
+				src.light(user, "<span class='alert'><b>[user]</b> lights [src] with [W]. Goddamn.</span>")
```

Comment:
this is just copy pasted from `candle`, any reason this item isn't a candle subtype?